benchmarks/helpers/main.ts

```diff
@@ -31,6 +31,10 @@ export async function runAllBenchmarks() {
   const allResults: BenchmarkResult[] = [];
 
   for (const [benchmark, benchmarks] of getRegisteredBenchmarks()) {
+    if (benchmarks.length === 0) {
```

Comment:
Could you please add a short comment on why we need to skip empty benchmarks? Seems to be an issue with benny.

Also I think this check would be better done in the `runBenchmark` function (which then should return `Summary | undefined`) and the return checked for undefined in this loop. That way the next person working with the code is likely to be aware of the empty benchmark issue.

But this is just a small script :grin: - so you can change that if you like or leave it like it is if you're in a hurry.

Reply:
> Also I think this check would be better done in the runBenchmark function (which then should return Summary | undefined) and the return checked for undefined in this loop. That way the next person working with the code is likely to be aware of the empty benchmark issue.

Also thought about this myself 😁 I'm going to do it this way. Btw, do you prefer `Summary | undefined` or `Summary | null`?

Reply:
For code I prefer `undefined` over `null`, as for undefined you can just write `return` instead of `return null`. But I wouldn't start a discussion over it you use what you prefer :grin: .

Reply:
I prefer ReScript 🤪

Reply:
Oh right that's worth a look too. Does ReScript have runtime type checking (and by extend runtime type information) built in or does it need 3rd part libraries like TS does?

Reply:
It's the same story as for TS. There's an option to generate encoders/decoders from type definition like [decco](https://github.com/reasonml-labs/decco) does, but this approach is discouraged by ReScript maintainers.

Personally, I like the idea to have runtime contracts only on the entries to your system and rely on static analysis inside of it. Also, having this layer with parsing and serializing is useful for transforming data into more convenient ReScript data structures.
Eg in TS we'd have a type:
```typescript
type Shape =
| { kind: "circle"; radius: number }
| { kind: "square"; x: number }
| { kind: "triangle"; x: number; y: number };
```
But in ReScript we'd use a variant for it
```rescript
type shape = Circle({radius: float}) | Square({x: float}) | Triangle({x: float, y: float})
```